ebean-core/src/main/java/io/ebeaninternal/api/BindParams.java

```diff
@@ -54,12 +54,11 @@ public void reset() {
     positionedParameters.clear();
   }
 
-  public int queryBindHash() {
-    int hc = namedParameters.hashCode();
+  public void queryBindHash(BindHash hash) {
+    hash.update(positionedParameters.size());
```

Comment:
all `queryBindHash` are changed. For collections, I changed the format also to append the size of the collection, otherwise you may run into collisions if you may have queries like these: `in("prop1", 1,2).in("prop2", 3)` `in("prop1", 1).in("prop2", 2, 3)` which have the same hash (also if MD5 is used)